indico/modules/events/editing/controllers/backend.py

```diff
@@ -250,3 +259,27 @@ def _process_PATCH(self, text, internal):
     def _process_DELETE(self):
         delete_revision_comment(self.comment)
         return '', 204
+
+
+class RHRevisionFilesExportZIP(RHContributionEditableRevisionBase):
+    def _check_revision_access(self):
+        return self._user_is_authorized_submitter() or self._user_is_authorized_editor()
+
+    def _process(self):
+        if not self.revision.files:
+            raise NotFound
+
+        temp_file = NamedTemporaryFile(suffix='indico.tmp', dir=config.TEMP_DIR)
+        with ZipFile(temp_file.name, 'w', allowZip64=True) as zip_handler:
+            for revision_file in self.revision.files:
+                file = revision_file.file
+                file_type = revision_file.file_type
+
+                with file.storage.get_local_path(file.storage_file_id) as filepath:
+                    filename = secure_filename(file.filename, 'file-{}'.format(file.id))
+                    folder_name = secure_filename(file_type.name, 'file-type-{}'.format(file_type.id))
+                    zip_handler.write(filepath.encode('utf-8'), os.path.join(folder_name, filename))
```

Comment:
i don't think we need/want `secure_filename` for the folder / filename used in the second arg. Just make sure that putting `../` and similar stuff there doesn't break anything (`secure_filename` strips much more, like any whitespace, non-ascii, etc).

Reply:
Hm, but `secure_filename` also makes sure that characters like `../` are stripped. I still would use it to assure security and not care about some characters being stripped.

Reply:
if you pass bytes to werkzeug's secure_filename it will only strip nasty stuff

Reply:
What do you mean exactly?

```python
In [70]: secure_filename(u'../łąe/as/asd/'.encode('utf-8'), 'test')                                                                                                                                                                                                                                                           
Out[70]: 'lae_as_asd'
```

That looks fine for me

Reply:
ah i didn't notice that the werkzeug function strips more stuff later with a regex..  so nevermind on that.

Reply:
anyway, since this is in a zipfile (opened on many different systems) maybe it's easier to stick with our `secure_filename`..